### PR TITLE
Task/bws154 initial user data script while creating instance

### PR DESCRIPTION
**Pull request by tbezrukava**
Снимок экрана 2022-03-20 в 20 21 31
Снимок экрана 2022-03-20 в 20 22 29

### CODE REVIEW

File: frontend/src/components/common/input/input.tsx [hunk 1]
```diff
@@ -26,12 +27,20 @@ const Input: FC<Props> = ({
...
+    <textarea
+      {...field}
+      rows={rows}
+      placeholder={placeholder}
+      className={styles.textarea}
+    />
+  ) : (
     <label className={styles.inputLabel}>
       <span className={styles.span}>{label}</span>
       <input
```
Comment (what1s1ove) [inline frontend/src/components/common/input/input.tsx]:
```suggestion
  return
    <label className={styles.inputLabel}>
      <span className={styles.span}>{label}</span>
       isTextarea ? (
    <textarea
      {...field}
      rows={rows}
      placeholder={placeholder}
      className={styles.textarea}
    />
  ) : ( <input
```

File: frontend/src/components/common/input/input.tsx [hunk 2]
```diff
@@ -26,12 +27,20 @@ const Input: FC<Props> = ({
...
   const hasError = Boolean(errors[name]);
+  const isTextarea = Boolean(rows);
 
-  return (
+  return isTextarea ? (
+    <textarea
+      {...field}
+      rows={rows}
+      placeholder={placeholder}
+      className={styles.textarea}
```
Comment (what1s1ove) [inline frontend/src/components/common/input/input.tsx]: we should have the same class for both inputs

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Comment by tbezrukava** [comment]
Снимок экрана 2022-03-20 в 21 22 15

File: frontend/src/components/common/input/input.tsx [hunk 3]
```diff
@@ -26,20 +27,32 @@ const Input: FC<Props> = ({
...
-        type={type}
-        placeholder={placeholder}
-        className={getValidClasses(hasError ? styles.inputError : styles.input)}
-      />
+      {isTextarea ? (
+        <textarea
+          {...field}
+          rows={rows}
+          placeholder={placeholder}
+          className={`${styles.input} ${styles.textarea}`}
```
Comment (what1s1ove) [inline frontend/src/components/common/input/input.tsx]: use getValidClasses
we also need to handle an error on the text-aria

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by Yaroslaww-1 — DISMISSED** [review]

File: frontend/src/components/sc-configurate-instance/components/create-instance-form/create-instance-form.tsx [hunk 4]
```diff
@@ -65,6 +65,17 @@ const CreateInstanceForm: FC = () => {
...
           </div>
+          <h3 className={styles.inputTitle}>User data</h3>
+          <div className={styles.inputWrapper}>
+            <Input
+              label=""
+              placeholder="(Optional)"
+              name={getNameOf<SCInstanceCreateRequestDto>('userData')}
+              control={control}
+              errors={errors}
+              rows={10}
```
Comment (junojei) [inline frontend/src/components/sc-configurate-instance/components/create-instance-form/create-instance-form.tsx:76]: A11?

**Review by Donymak — DISMISSED** [review]

File: frontend/src/components/sc-configurate-instance/components/create-instance-form/create-instance-form.tsx (hunk 4 above)
Comment (what1s1ove) [reply frontend/src/components/sc-configurate-instance/components/create-instance-form/create-instance-form.tsx:76]: it's okay for now

**Review by ania-leichenko — APPROVED** [review]

**Review by junojei — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**